framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/builder/AbstractKieServerWithDatabaseScenarioBuilder.java

```diff
@@ -0,0 +1,170 @@
+
+    @Override
+    public KieServerWithDatabaseScenario getDeploymentScenarioInstance() {
+        return new KieServerWithDatabaseScenarioImpl(kieApp, deploySSO);
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder deploySso() {
+        deploySSO = true;
+        SsoClient ssoClient = new SsoClient();
+        ssoClient.setName("workbench-client");
+        ssoClient.setSecret("workbench-secret");
+        kieApp.getSpec().getObjects().getConsole().setSsoClient(ssoClient);
+
+        Server[] servers = kieApp.getSpec().getObjects().getServers();
+        for (int i = 0; i < servers.length; i++) {
+            ssoClient = new SsoClient();
+            ssoClient.setName("kie-server-" + i + "-client");
+            ssoClient.setSecret("kie-server-" + i + "-secret");
+            servers[i].setSsoClient(ssoClient);
+        }
+        return this;
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withKieServerId(String kieServerId) {
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            server.addEnv(new Env(ImageEnvVariables.KIE_SERVER_ID, kieServerId));
+        }
+        return this;
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withHttpKieServerHostname(String hostname) {
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            server.addEnv(new Env(ImageEnvVariables.HOSTNAME_HTTP, hostname));
+        }
+        return this;
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withHttpsKieServerHostname(String hostname) {
+        for (Server server : kieApp.getSpec().getObjects().getServers()) {
+            server.addEnv(new Env(ImageEnvVariables.HOSTNAME_HTTPS, hostname));
+        }
+        return this;
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withExternalLdap(LdapSettings ldapSettings) {
+        Ldap ldap = LdapSettingsMapper.toLdapModel(ldapSettings);
+        Auth auth = new Auth();
+        auth.setLdap(ldap);
+        kieApp.getSpec().setAuth(auth);
+        return this;
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withInternalLdap(LdapSettings ldapSettings) {
+        setAsyncExternalDeployment(ExternalDeploymentID.LDAP);
+        return withExternalLdap(ldapSettings);
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withDeploymentScenarioListener(DeploymentScenarioListener<KieServerWithDatabaseScenario> deploymentScenarioListener) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withInternalMavenRepo(boolean waitForRunning) {
+        setAsyncExternalDeployment(ExternalDeploymentID.MAVEN_REPOSITORY);
+        return this;
+    }
+
+    @Override
+    public KieServerWithDatabaseScenarioBuilder withContainerDeployment(String kieContainerDeployment) {
```

Comment:
It seems that we don't support unmanaged Kie server deployments apart from S2I build for operator.
In such case IMHO this method doesn't have meaning as Kie server always retrieves configuration from controller. If so please throw an exception for this method as it is unsupported.

That would mean that we should skip KieServerWithBuiltKjarIntegrationTest for operator as it is irrelevant.

Reply:
Updated withContainerDeployment.

Reply:
Please update also KieServerWithBuiltKjarIntegrationTest with OperatorNotSupported category, so it is skipped.